### ISSUE TITLE
Airflow Chart - How to pass Personal Access Token as a secret for git clone of DAGs

### ISSUE THREAD

**Issue by pankajkoti** [bitnami/charts#3455]
The documentation says to use the following:
If you use a private repository from GitHub, a possible option to clone the files is using a Personal Access Token and using it as part of the URL: https://USERNAME:PERSONAL_ACCESS_TOKEN@github.com/USERNAME/REPOSITORY

How can I pass this PERSONAL_ACCESS_TOKEN as a secret to this chart?

**Comment by marcosbc** [comment]
Hi @pankajkoti, you could make use of `extraEnvVarsCM` to specify an environment variable for the GitHub access token, i.e. naming it `GITHUB_ACCESS_TOKEN`.

Then, deploying the chart with the option below should work properly:

```
--set 'airflow.cloneDagFilesFromGit.repository=https://myuser:${GITHUB_ACCESS_TOKEN}@github.com/USER/REPO'
```

**Comment by pankajkoti** [comment]
Hi marcosbc, yes, but then how to pass the value as a secret in extraEnvVarsCM for the key 'GITHUB_ACCESS_TOKEN'

**Comment by marcosbc** [comment]
Sorry, my bad, I should have referred to `extraEnvVarsSecret`. Pass the name of your secret containing the `GITHUB_ACCESS_TOKEN` environment variable to `airflow.gitCloneContainer.extraEnvVarsSecret` and it should work.

**Comment by pankajkoti** [comment]
Okay, I tried this:
`airflow:
    cloneDagFilesFromGit:
      enabled: true
      repository: https://<my_user_name_here>:${GITLAB_ACCESS_TOKEN}@<my_repository_here>
      branch: master
      path: etl/dags
    gitCloneContainer:
      extraEnvVarsSecret: gitlab-token`

And my secrets file is:
`apiVersion: v1
kind: Secret
metadata:
  name: gitlab-token
type: Opaque
data:
  gitlab_access_token: <my_secret_here>
`

I also tried creating the secret like this:
`apiVersion: v1
kind: Secret
metadata:
  name: gitlab-token
type: Opaque
data:
  GITLAB_ACCESS_TOKEN: ayixhzJXk8RgEjTekUca
`

But, I get access denied here.

It's working if I directly pass the token in the url, but not with the secret env variable way. Am I missing something here?
